.github/workflows/ci.yaml

```diff
@@ -60,3 +60,56 @@ jobs:
               atmosphere.common.flux \
               atmosphere.common.cert_manager \
               atmosphere.common.metal3
+
+  metrics-server:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - uses: helm/kind-action@a1b0e391336a6ee6713a0583f8c6240d70863de3 # v1.12.0
+      - uses: astral-sh/setup-uv@d4b2f3b6ecc6e67c4457f6d3e41ec42d3d0fcb86 # v5.4.2
+      - run: uv run ansible-galaxy collection install ${{ github.workspace }}
+      - name: Create metrics-values.yml
+        run: |
+          cat <<EOF > metrics-values.yml
+          metrics_server_helm_release_namespace: kube-system
+          metrics_server_helm_values:
+            args:
+              - --kubelet-insecure-tls
+          EOF
+
+      - name: Apply metrics-server HelmRelease
+        run: |
+          uv run ansible-playbook \
+            --connection=local \
+            --extra-vars target=localhost \
+            --extra-vars @metrics-values.yml \
+              atmosphere.common.flux \
+              atmosphere.common.metrics_server
+      - name: Wait for metrics-server HelmRelease to be ready
+        run: |
+          echo "Waiting for HelmRelease/metrics-server to be ready..."
+          for i in {1..30}; do
+            status=$(kubectl get helmrelease metrics-server -n kube-system -o jsonpath='{.status.conditions[?(@.type=="Ready")].status}' 2>/dev/null || echo "")
+            if [ "$status" == "True" ]; then
+              echo "HelmRelease is ready."
+              break
+            fi
+            echo "Still waiting... ($i/30)"
+            sleep 10
+          done
+          if [ "$status" != "True" ]; then
+            echo "HelmRelease did not become ready in time."
+            kubectl get helmrelease metrics-server -n kube-system -o yaml
+            exit 1
+          fi
```

Comment:
This can be replaced with a kubectl wait since there is a condition when things get deployed.